msticpy/data/drivers/sumologic_driver.py

```diff
@@ -0,0 +1,541 @@
+                    help_uri="https://msticpy.readthedocs.io/en/latest/DataProviders.html",
+                ) from err
+        elif status["state"] == "DONE GATHERING RESULTS":
+            # Aggregated results
+            count = status["recordCount"]
+            limit2 = (
+                count if count < limit and count != 0 else limit
+            )  # compensate bad limit check
+            try:
+                result = self.service.search_job_records(searchjob, limit=limit2)
+                return result["records"]
+            except HTTPError as err:
+                raise MsticpyConnectionError(
+                    f"Communication error connecting to Sumologic: {err}",
+                    title="Sumologic search_job_records",
+                    help_uri="https://msticpy.readthedocs.io/en/latest/DataProviders.html",
+                ) from err
+            except Exception as err:
+                raise MsticpyConnectionError(
+                    f"Failed to get search records: {err}",
+                    title="Sumologic connection",
+                    help_uri="https://msticpy.readthedocs.io/en/latest/DataProviders.html",
+                ) from err
+
+        return []
+
+    def query(
+        self, query: str, query_source: QuerySource = None, **kwargs
+    ) -> Union[pd.DataFrame, Any]:
+        """
+        Execute Sumologic query and retrieve results.
+
+        Parameters
+        ----------
+        query : str
+            Sumologic query to execute
+
+        Other Parameters
+        ----------------
+        kwargs :
+            Are passed to Sumologic
+            days: Search the past X days.
+            start_time: A datetime() object representing the start of the search
+                    window. If used without end_time, the end of the search
+                    window is the current time.
+            end_time: A datetime() object representing the end of the search window.
+                  If used without start_time, the search start will be the earliest
+                  time in the index.
+            timeZone: timezone used for time range search
+            byReceiptTime:  if time reference should used _receiptTime (time when Sumologic
+                        got message) instead of _messageTime (time present in log message).
+            limit: An integer describing the max number of search results to return.
+            forceMessagesResults: Force results to be raw messages even if aggregated query.
+            verbosity: Provide more verbose state. from 0 least verbose to 4 most one.
+            normalize: If set to True, fields containing structures (i.e. subfields)
+                   will be flattened such that each field has it's own column in
+                   the dataframe. If False, there will be a single column for the
+                   structure, with a JSON string encoding all the contents.
+            exporting: Export result to file.
```

Comment:
Do you need an "exporting" flag as well as "export_path"? If export_path is None then just ignore the export part